src/qrules/settings.py

```diff
@@ -313,23 +318,21 @@ def __positive_int_domain(
     return _int_domain(0, max(values))
 
 
-def _halves_domain(start: float, stop: float) -> list[float]:
-    if start % 0.5 != 0.0:
+def _halves_domain(start: Fraction, stop: Fraction) -> list[Fraction]:
```

Comment:
Rewrite such that
```suggestion
def _halves_domain(start: SupportsFloat, stop: SupportsFloat) -> list[Fraction]:
```

probably with

```python
    start = Fraction(start)
    stop = Fraction(stop)
```

Reply:
Ah wait, `arange()` already does this! So you don't need to cast.

Reply:
As of now arange doesn't cast and `halves_domain` converts to `Fraction` as originally recommended, because this is also used in the input validation.

Reply:
`SupportsFloat` doesn't work nicely with `Fraction` in mypy. `float` is used instead, as before.